Release notes v21.2.7 (#13217)
cockroachdb/docs: _includes/releases/v21.2/v21.2.7.md

<h3 id="v21-2-7-enterprise-edition-changes">Enterprise edition changes</h3>

- Currently executing schedules are cancelled immediately when the jobs scheduler is disabled.  Release justification: stability improvement. [#77313][#77313]

<h3 id="v21-2-7-sql-language-changes">SQL language changes</h3>

- When dropping a user that has default [privileges](../v21.2/security-reference/authorization.html#privileges), the error message now includes which database and schema the default privileges are defined in.  Additionally, a hint is now given to show exactly how to remove the default privileges. [#77142][#77142]
- Previously, `crdb_internal.default_privileges` would incorrectly show default [privileges](../v21.2/security-reference/authorization.html#privileges) for databases where the default privilege was not actually defined. [#77304][#77304]

<h3 id="v21-2-7-operational-changes">Operational changes</h3>

- Operators who wish to access HTTP endpoints of a cluster through a proxy can now request specific node IDs through a `remote_node_id` query parameter or cookie with the value set to the node IDs they would like to proxy the connection to. [#76694][#76694]

> - Added the [cluster setting](../v21.2/cluster-settings.html) `bulkio.backup.resolve_destination_in_job.enabled`, which can be used to delay resolution of backup's destination until the job starts running. [#76816][#76816]

Reviewer: ```suggestion
- Added the [cluster setting](../v21.2/cluster-settings.html) `bulkio.backup.resolve_destination_in_job.enabled`, which can be used to delay resolution of a backup's destination until the job starts running. [#76816][#76816]
```

<h3 id="v21-2-7-db-console-changes">DB Console changes</h3>

> - Open SQL Transactions and Active SQL Transactions are now downsampled using MAX instead of AVG and will more accurately reflect narrow spikes in transaction counts when looking and downsampled data. [#76688][#76688]

Reviewer: ```suggestion
- Open SQL Transactions and Active SQL Transactions are now downsampled using `MAX` instead of `AVG` and will more accurately reflect narrow spikes in transaction counts when looking and downsampled data. [#76688][#76688]
```

> - DB Console requests can be routed to arbitrary nodes in the cluster. Users can select a node from a dropdown in the Advanced Debug page of the DB Console UI to route their UI to that node. Manually initiated requests can either add a `remote_node_id` query param to their request or set a `remote_node_id` HTTP cookie in order to manage the routing of their request. [#76694][#76694]

Reviewer: ```suggestion
- DB Console requests can be routed to arbitrary nodes in the cluster. Users can select a node from a dropdown in the **Advanced Debug** page of the DB Console UI to route their UI to that node. Manually initiated requests can either add a `remote_node_id` query param to their request or set a `remote_node_id` HTTP cookie in order to manage the routing of their request. [#76694][#76694]
```

> - Add long loading messages to SQL Activity pages. [#77008][#77008]

Reviewer: ```suggestion
- Add long loading messages to **SQL Activity** pages. [#77008][#77008]
```

> - Remove stray parenthesis at the end of the duration time for a succeeded job. It had been accidentally introduced to unreleased master and a 21.2 backport.  Release justification: Category 2, UI bug fix [#77444][#77444]

Reviewer: ```suggestion
- Removed a stray parenthesis at the end of the duration time for a succeeded job. [#77444][#77444]
```

- Fixed a bug that could corrupt indexes containing [virtual columns](../v21.2/computed-columns.html) or [expressions](../v21.2/expression-indexes.html). The bug only occurred when the index's table had a foreign key reference to another table with an `ON DELETE CASCADE` action, and a row was deleted in the referenced table. This bug was present since virtual columns were added in version v21.1.0. [#77053][#77053]
- [Changefeeds](../v21.2/changefeed-sinks.html) retry instead of fail on RPC send failure. [#77069][#77069]
- Fixed a bug that caused the **Open Transactions** chart on the [**Metrics**](../v21.2/ui-overview-dashboard.html) page to constantly increase for empty transactions. [#77236][#77236]
- Fixed a bug that could interfere with a system table migration. [#77309][#77309]
- The content type header for the HTTP log sink is now set to `application/json` if the format of the log output is `JSON`. [#77341][#77341]
- Fixed a bug where draining nodes in a cluster without shutting them down could stall foreground traffic in the cluster. [#77490][#77490]

<h3 id="v21-2-7-performance-improvements">Performance improvements</h3>

> - Fixed a bug in the histogram estimation code that could cause the optimizer to think a scan of a multi-column index would produce 0 rows, when in fact it would produce many rows. This could cause the optimizer to choose a suboptimal plan. It is now less likely for the optimizer to choose a suboptimal plan when multiple multi-column indexes are available. [#76555][#76555]

Reviewer: ```suggestion
- Fixed a bug in the histogram estimation code that could cause the [optimizer](../v21.2/cost-based-optimizer.html) to think a scan of a multi-column index would produce 0 rows, when in fact it would produce many rows. This could cause the optimizer to choose a suboptimal plan. It is now less likely for the optimizer to choose a suboptimal plan when multiple multi-column indexes are available. [#76555][#76555]
```
